src/solver/opamActionGraph.mli

```diff
@@ -36,16 +36,18 @@ module type SIG = sig
       There is no guarantee however that the resulting graph is acyclic. *)
   val reduce: t -> t
 
-  (** Expand install actions, adding a build action preceding them. *)
-  val explicit: t -> t
-end
+  (** Expand install actions, adding a build action preceding them.
+      The argument [concrete] is a function that should return `true`
+      on package where the `remove` action is a `noop`
+      (such as `conf-*` package). *)
+  val explicit: ?noop_remove:(package -> bool) -> t -> t end
```

Comment:
My comments is indeed misleading. the `noop-remove` really mean "the removal of this package won't touch the prefix". It might have effect on the environment, the previous patch https://github.com/ocaml/opam/pull/2593/commits/0d1275843f841ff77c3efcb55a01e2c648aa944f already handles that. What I am trying to detect here is : whether we might build a package before to remove/install its predecessors from the prefix (in all circumstance, it is build in an environment where the variable installed by its predecessors are removed). Turning reinstall into install for conf-* package does not seem sufficient to detect that AFAIK.